.github/workflows/autobuild.yml

```diff
@@ -65,92 +93,190 @@ jobs:
   release_assets:
     name:                           Build assets for ${{ matrix.config.config_name }}
     needs:                          create_release
-    runs-on:                        ${{ matrix.config.building_on_os }}
     strategy:
       fail-fast:                    false
       matrix: # Think of this like a foreach loop. Basically runs the steps with every combination of the contents of this. More info: https://docs.github.com/en/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
         config:
-           - config_name:           Linux (normal)
+           - config_name:           AndroidAPK (artifact+codeql)
+             target_os:             android
+             building_on_os:        ubuntu-20.04
+             cmd1_prebuild:         "sh android/autobuild/autobuild_apk_1_prepare.sh "
+             cmd2_build:            "sh android/autobuild/autobuild_apk_2_build.sh "
+             cmd3_postbuild:        "sh android/autobuild/autobuild_apk_3_copy_files.sh "
+             uses_codeql:           true
+
+           - config_name:           Linux (codeQL)
              target_os:             linux
-             building_on_os:        ubuntu-18.04
-             asset_path:            deploy/Jamulus_amd64.deb
-             asset_name:            jamulus_${{ needs.create_release.outputs.version_name }}_ubuntu_amd64.deb
-           - config_name:           Linux (headless)
+             building_on_os:        ubuntu-20.04
+             cmd1_prebuild:         "sh linux/autobuild/codeQL/auto_1_prepare.sh ${{ needs.create_release.outputs.x_github_workspace }}"
```

Comment:
We seem to have a lot of scripts to maintain now.  How much of the code is duplicated between them?  Have they been refactored to make the reply to that as close to "none" as possible?

Reply:
There are not more than before, but less scattered and uniformly splitted in 3 steps (to see what effort goes into building and what into preparation).

My idea was to first set an overall structure in this PR, and then be able to clean stages separately. (e.g. some latforms still rely on additional scripts outside the autobuild folders)